File: app/channels/conversation_channel.rb
# Diffuse les Turbo Stream de remplacement de messages pour une conversation donnee.
# Les clients abonnes recoivent chaque chunk de l'IA en temps reel et Turbo
# remplace le DOM de la bulle assistant.
class ConversationChannel < ApplicationCable::Channel
  def subscribed
    conversation = current_user.conversations.find_by(id: params[:conversation_id])
    return reject unless conversation

    stream_from "conversation_#{conversation.id}"
  end

  def unsubscribed
    # cleanup si necessaire
  end
end

File: app/javascript/channels/conversation_channel.js
// Canal de streaming par conversation.
// Chaque client s'abonne a `conversation_#{id}` et recoit les turbo_stream
// de remplacement de messages (utilise par les broadcasts du job streaming).
import consumer from "consumer"

consumer.subscriptions.create(
  { channel: "ConversationChannel", conversation_id: window.location.pathname.split("/").pop() },
  {
    connected() {
      // Subscription OK
    },
    received(data) {
      // Le data est un fragment HTML turbo_stream (turbo_stream.replace).
      // Turbo l'intercepte et remplace le DOM cible.
      if (data.html) {
        Turbo.renderStreamMessage(data.html)
      }
    }
  }
)
